lib/activerecord-bitemporal/scope.rb

```diff
@@ -141,6 +141,18 @@ def bitemporal_value=(value)
     end
   end
 
+  module CollectionProxy
+    # Delegate ActiveRecord::Bitemporal::Relation#bitemporal_value
+    # @see https://github.com/rails/rails/blob/v7.1.3.4/activerecord/lib/active_record/associations/collection_proxy.rb#L1115-L1124
+    def bitemporal_value
+      scope.bitemporal_value
+    end
+
+    def bitemporal_value=(value)
+      scope.bitemporal_value = value
+    end
```

Comment:
I thought `delegate` could use it. This works fine in my local environment.
```suggestion
    delegate :bitemporal_value, :bitemporal_value=, to: :scope
```

`delegate` dynamically generates similar methods.
see: https://github.com/rails/rails/blob/v7.1.3.4/activesupport/lib/active_support/core_ext/module/delegation.rb#L171

Reply:
Agreed, but I'm wondering if it's safe to not call it within `included` or `prepended`. For instance, in Rails, `delegate` seems to be called within `included`.
https://github.com/rails/rails/blob/v7.1.3.4/actiontext/lib/action_text/rendering.rb#L12

Indeed, looking at the (current) implementation, it seems like there would be safe, but I'm not sure if that is the intended design.

Reply:
Ah, I see. That code in Rails is special because it delegates to `class`.

There is code to optimize when the delegate is a `Module` or a `class`.
https://github.com/rails/rails/blob/v7.1.3.4/activesupport/lib/active_support/core_ext/module/delegation.rb#L210-L214
This implementation was introduced in https://github.com/rails/rails/pull/46875.
Therefore, if the delegate is a `class`, the definition will change depending on whether or not the method exists at the time of delegation.

In this case, using `delegate` seems to work fine, but I thought it was fine as is.

> Indeed, looking at the (current) implementation, it seems like there would be safe, but I'm not sure if that is the intended design.

If the current design were to be changed, it would be a big incompatibility. Therefore, if changes are to be made, there is no need to worry as a deprecation period and migration method will be provided.

Reply:
After internal discussion, I have decided to accept this suggestion.
I also tried using `prepended`, but it didn't work as expected. When using the `prepended` block, methods are defined directly on the delegate class, so the `ActiveRecord::Bitemporal::Relation` that was prepended before takes precedence.